scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/ScalametaTokenizer.scala

```diff
@@ -177,26 +179,32 @@ class ScalametaTokenizer(input: Input, dialect: Dialect) {
   private def getToken(curr: LegacyTokenData): Token = {
     (curr.token: @scala.annotation.switch) match {
       case IDENTIFIER => Token.Ident(input, dialect, curr.offset, curr.endOffset, curr.strVal)
-      case INTLIT => Token.Constant.Int(input, dialect, curr.offset, curr.endOffset, curr.intVal)
-      case LONGLIT => Token.Constant.Long(input, dialect, curr.offset, curr.endOffset, curr.longVal)
-      case FLOATLIT => Token.Constant
-          .Float(input, dialect, curr.offset, curr.endOffset, curr.floatVal)
-      case DOUBLELIT => Token.Constant
-          .Double(input, dialect, curr.offset, curr.endOffset, curr.doubleVal)
+      case INTLIT => curr.intVal
+          .fold(getInvalid(curr, _), Token.Constant.Int(input, dialect, curr.offset, curr.endOffset, _))
+      case LONGLIT => curr.longVal
+          .fold(getInvalid(curr, _), Token.Constant.Long(input, dialect, curr.offset, curr.endOffset, _))
+      case FLOATLIT => curr.floatVal.fold(
+          getInvalid(curr, _),
+          Token.Constant.Float(input, dialect, curr.offset, curr.endOffset, _)
+        )
+      case DOUBLELIT => curr.doubleVal.fold(
+          getInvalid(curr, _),
+          Token.Constant.Double(input, dialect, curr.offset, curr.endOffset, _)
+        )
       case CHARLIT => Token.Constant.Char(input, dialect, curr.offset, curr.endOffset, curr.charVal)
       case SYMBOLLIT => Token.Constant
           .Symbol(input, dialect, curr.offset, curr.endOffset, scala.Symbol(curr.strVal))
       case STRINGLIT => Token.Constant
           .String(input, dialect, curr.offset, curr.endOffset, curr.strVal)
-      case STRINGPART => unreachable
+//      case STRINGPART => unreachable
       case TRUE => Token.KwTrue(input, dialect, curr.offset)
       case FALSE => Token.KwFalse(input, dialect, curr.offset)
       case NULL => Token.KwNull(input, dialect, curr.offset)
 
       case INTERPOLATIONID => Token.Interpolation
           .Id(input, dialect, curr.offset, curr.endOffset, curr.strVal)
       case XMLLIT => getXmlPart(curr)
-      case XMLLITEND => unreachable
+//      case XMLLITEND => unreachable
```

Comment:
```suggestion
```
and also other below?

Reply:
do you mean to remove the lines completely? i kept them to indicate that these particular tokens should not be obtained via this method, and it's not an oversight they are not here.

Reply:
Should we just add a comment for all those at the end? It's fine either way

Reply:
done